### ISSUE TITLE
no-duplicate-default-export

### ISSUE THREAD

**Issue by benmosher** [benmosher/eslint-plugin-import#22]
Flag subsequent `export default ...` statements if more than one is encountered in a given file.

**Comment by benmosher** [comment]
fixed in `master`, also duplicate named exports (see `import/export` rule)

**Comment by AndrewLeedham** [comment]
@benmosher Came across an issue with this feature, thought I would open a discussion here:

The following example causes 3 `error  Multiple default exports  import/export` errors, but there is only technically 1 default export the other 2 are overloads.
```TypeScript
export default function foo(param: string): boolean;
export default function foo(param: string, param1: number): boolean;
export default function foo(param: string, param1?: number): boolean {
  return param && param1;
}
```
There seems to be some code for checking overloads: https://github.com/benmosher/eslint-plugin-import/blob/master/src/rules/export.js#L37-L40 but this does not seem to catch this case. I added some debug logs in and the rough contents of namespace is:
```
root: {
  default: [
    {
       type: 'ExportDefaultDeclaration'
    },
    {
       type: 'ExportDefaultDeclaration'
    },
    {
       type: 'ExportDefaultDeclaration'
    }
  ]
}
```

I am running this in the following environment:
* **eslint**: 5.16.0
* **eslint-plugin-import**: 2.19.1
* **eslint-import-resolver-typescript**: 2.0.0
* **typescript**: 3.7.2
* **@typescript-eslint/parser**: 2.13.0

**Comment by ljharb** [comment]
Please file a new issue for this (altho i think there already is one)

**Comment by AndrewLeedham** [comment]
This one? https://github.com/benmosher/eslint-plugin-import/issues/1357

**Comment by ljharb** [comment]
That’s fixed in v2.19, so perhaps a new issue is best.
